compiler/src/iree/compiler/Dialect/HAL/Target/ROCM/ROCMTarget.cpp

```diff
@@ -177,8 +194,12 @@ class ROCMTargetBackend final : public TargetBackend {
 
     // Serialize hsaco kernel into the binary that we will embed in the
     // final flatbuffer.
-    std::string targetISA = translateModuleToISA(*llvmModule, *targetMachine);
-    std::string targetHSACO = createHsaco(targetISA, libraryName);
+    if (dumpIsa) {
+      std::string targetISA = translateModuleToISA(*llvmModule, *targetMachine);
+      llvm::dbgs() << targetISA;
```

Comment:
there's a mechanism that gives you access to these outputs in a way you can integrate into pipelines - unless you really need it mixed into the debug output stream you should use that - then you can get the assembly files directly (one per dispatch) and pass them to other tools/diff them/etc.

See this which dumps SPIRV assembly (you should be able to just copy that line) if you pass `--iree-hal-dump-executable-intermediates-to=some/path/`:
https://github.com/google/iree/blob/41a2cebe9560fda8a3ef171bb2dd5f51434897f2/compiler/src/iree/compiler/Dialect/HAL/Target/WebGPU/WebGPUTarget.cpp#L153-L154

Reply:
I see and what will I have to do for this part?
https://github.com/google/iree/blob/41a2cebe9560fda8a3ef171bb2dd5f51434897f2/compiler/src/iree/compiler/Dialect/HAL/Target/WebGPU/WebGPUTarget.cpp#L149-L152

Reply:
I think I figure it out I dont need the disassembley stuff right?